src/activities/facebook/facebook_location_history.rs

```diff
@@ -28,21 +29,42 @@ pub struct Coordinate {
 #[allow(non_snake_case)]
 impl LocationHistory {
     pub fn saveToDb(&self,conn: &Connection) -> Result<(), rusqlite::Error> {
-        for elem in self.location_history_v2.iter() {
-            conn.execute(
-                "INSERT into facebook_location_history
-                (time, name, latitude, longitude)
-                values(?1, $2, $3, $4)",
-                params![
-                    elem.creation_timestamp.to_string(),
-                    elem.name,
-                    elem.coordinate.latitude.to_string(),
-                    elem.coordinate.longitude.to_string()
-                ],
-            )
-            .map_err(|err| println!("{:?}", err))
-            .ok();
-        }
+        match &self.location_history_v2 {
+            None => None,
+            Some(x) => Some (for elem in x.iter() {
```

Comment:
I appreciate your work and sorry for my past redundant code. I am now thinking that it can be refactored like below(sorry i haven't run this below code yet)

```
        match &self.location_history_v2 {
            None => None,
            Some(elems) => elems.iter().map(|elem| {
                conn.execute(
                    "INSERT into facebook_location_history
                    (time, name, latitude, longitude)
                    values(?1, $2, $3, $4)",
                    params![
                        elem.creation_timestamp.to_string(),
                        elem.name,
                        elem.coordinate.latitude.to_string(),
                        elem.coordinate.longitude.to_string()
                    ],
                )
                .map_err(|err| println!("{:?}", err))?
            }),
        };
```

Reply:
I tried this code and there are some errors. I used the compiler to try to correct them, but even when there is no error, it doesn't work. I had the same problem with the previous code using `elems.iter().map(|elem| {}` method and that is why I changed it. Maybe I am doing it wrong.

Reply:
@GawenMonnot Thanks for your test. How about this code? I've checked with `dummy` data.

```
        match &self.location_history_v2 {
            Some(x) => Some (x.iter().for_each(|elem| {
                conn.execute(
                    "INSERT into facebook_location_history
                    (time, name, latitude, longitude)
                    values(?1, $2, $3, $4)",
                    params![
                        elem.creation_timestamp.to_string(),
                        elem.name,
                        elem.coordinate.latitude.to_string(),
                        elem.coordinate.longitude.to_string()
                    ],
                )
                .map_err(|err| println!("{:?}", err))
                .ok();
            })),
            None => None,
        };
```

Reply:
@chansuke Yes, this one works for me! Thank you very much. Should I edit my commit, or maybe you want to do it by yourself ?

Reply:
@GawenMonnot You can fix like below and please make sure to run `cargo clippy` command before you push the commit 👍 
```
        match &self.location_history_v2 {
            Some(x) => x.iter().for_each(|elem| {
                conn.execute(
                    "INSERT into facebook_location_history
                    (time, name, latitude, longitude)
                    values(?1, $2, $3, $4)",
                    params![
                        elem.creation_timestamp.to_string(),
                        elem.name,
                        elem.coordinate.latitude.to_string(),
                        elem.coordinate.longitude.to_string()
                    ],
                )
                .map_err(|err| println!("{:?}", err))
                .ok();
            }),
            None => println!("No data"), // You can change the sentence
        };
```

Reply:
@chansuke It should be done now. Sorry for the different commits : at first, I made a mistake and I changed the wrong file (device_location.rs).

Reply:
Thanks